libflux/src/flux/ast/mod.rs

```diff
@@ -416,7 +416,7 @@ pub struct Block {
 // BadStmt is a placeholder for statements for which no correct statement nodes
 // can be created.
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
-#[serde(rename = "BadStatement", tag = "type")]
```

Comment:
Where is this unneeded? If it is being duplicated, it needs to be removed from the other area. I found a few of these last time I looked through here. The tag is needed on this one to ensure deserialization works correctly.

Reply:
The other area is the `Statement` enum, since `BadStatement` never appears by itself, only as a variant within that enum, I believe that it is safe to remove.

Reply:
The test also roundtrips everything, including `BadStmt`, so it appears to work correctly.

Reply:
Ah, maybe I should put the `rename` back in, but I believe that `tag` is unneeded.

Reply:
Ah thanks for clarifying. Your comment reminded me that what I remembered was the other way around. It's the `Statement` that needs to have the tag not the individual element.

With that, I think you can remove the entire annotation. It seems to be what was done in d1c3353fe339df82253c872ea1e19ba802c1d39c for a similar problem.

Reply:
I think it still needs to be renamed, since the Go struct is `BadStatement`, and we need to be able to deserialize in Go, or at least we're supposed to be able to.

We didn't have a serialization test for `BadStatement` so I added one.